core/src/main/java/org/ldaptive/auth/ext/ActiveDirectoryAuthenticationResponseHandler.java

```diff
@@ -115,9 +117,6 @@ public Period getExpirationPeriod()
    */
   public void setExpirationPeriod(final Period period)
   {
-    if (period == null) {
-      throw new IllegalArgumentException("Expiration cannot be null");
-    }
```

Comment:
Why did you remove this null check? Seems unrelated to the refactoring described in the issue.